train/compute/python/tools/trace_link.py

```diff
@@ -375,23 +835,36 @@ def approximate_match(kineto_et_events, et_nodes):
 
     logger.setLevel(args.log_level)
 
-    et_nodes, kineto_et_events = trace_analysis(
-        args.et_file, args.kineto_file, args.annotation
-    )
+    et_gpu_ops_per_cpu_op_id = {}
+    (
+        et_nodes,
+        kineto_et_ops,
+        kineto_ac2g_s_ops,
+        kineto_ac2g_f_ops,
+        kineto_cpu_launcher_ops,
+        kineto_gpu_ops,
+    ) = trace_analysis(args.et_file, args.kineto_file, args.annotation)
 
     if args.exact_match:
-        et_enhanced_duration = exact_match(kineto_et_events, et_nodes)
+        (
+            et_enhanced_duration,
+            et_enhanced_timestamp,
+            et_gpu_ops_per_cpu_op_id,
+        ) = exact_match(
+            kineto_et_ops,
+            kineto_ac2g_s_ops,
+            kineto_ac2g_f_ops,
+            kineto_cpu_launcher_ops,
+            kineto_gpu_ops,
+            et_nodes,
+        )
+        # If linking works, add duration time to each ET node and dump as ET_plus
+        dump_et_file(
+            et_enhanced_duration, et_enhanced_timestamp, et_gpu_ops_per_cpu_op_id
+        )
     else:
-        et_enhanced_duration = approximate_match(kineto_et_events, et_nodes)
-
-    # If linking works, add duration time to each ET node and dump as ET_plus
-    if et_enhanced_duration:
-        et = read_dictionary_from_json_file(args.et_file)
-        for node in et["nodes"]:
-            if node["id"] in et_enhanced_duration:
-                node["dur"] = et_enhanced_duration[node["id"]]
-
-        et_plus_file = args.et_file.replace(".json", "_plus.json")
-        logger.info(f"Enhanced execution trace dumped to {et_plus_file}")
-
-        write_dictionary_to_json_file(et_plus_file, et)
+        logger.info(
+            "This script works only with the exact match mode,"
+            + "please add the --exact-match flag to your script and run it again"
+        )
+        assert False
```

Comment:
@JoongunPark  Please replace this with raising an exception.

Reply:
@TaekyungHeo Update it as requested.